_rules/descriptive-link-5effbb.md

```diff
@@ -0,0 +1,292 @@
+		<ul>
+			<li><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm"> HTML </a></li>
+			<li>
+				<a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73">
+					EPUB
+				</a>
+			</li>
+			<li><a href="https://www.gutenberg.org/files/4300/4300-0.txt"> Plain text </a></li>
+		</ul>
+	</li>
+</ul>
+```
+
+#### Passed Example 6
+
+The context provided by the table header and the [accessible name][] (from the link's text) describe the purpose of the links.
+
+```html
+<table>
+	<tr>
+		<th colspan="3">Ulysses</th>
+	</tr>
+	<tr>
+		<td><a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">HTML</a></td>
+		<td>
+			<a href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73"
+				>EPUB</a
+			>
+		</td>
+		<td><a href="https://www.gutenberg.org/files/4300/4300-0.txt">Plain text</a></td>
+	</tr>
+</table>
+```
+
+#### Passed Example 7
+
+The [accessible name][] describes the purpose of the link.
+
+```html
+<p id="instructions">Click on the arrow to go to the main content.</p>
+<svg xmlns="http://www.w3.org/2000/svg" x="0" y="0">
+	<a href="#main" aria-labelledby="instructions">
+		<path
+			style="fill:#1E201D;"
+			d="M21.205,5.007c-0.429-0.444-1.143-0.444-1.587,0c-0.429,0.429-0.429,1.143,0,1.571l8.047,8.047H1.111
+			C0.492,14.626,0,15.118,0,15.737c0,0.619,0.492,1.127,1.111,1.127h26.554l-8.047,8.032c-0.429,0.444-0.429,1.159,0,1.587
+			c0.444,0.444,1.159,0.444,1.587,0l9.952-9.952c0.444-0.429,0.444-1.143,0-1.571L21.205,5.007z"
+		/>
+	</a>
+</svg>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<a href="#desc">More</a>
+
+<p id="desc">This product consists of several web pages.</p>
+```
+
+#### Failed Example 2
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<div role="link" tabindex="0" onclick="document.location+='#main'">More</div>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 3
+
+The [accessible name][] (from the link's text), together with the absence of [programmatically determined link context][], does not describe the purpose of the link.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" x="0" y="0">
+	<a href="#main">
+		<text x="20" y="20">
+			Go
+		</text>
+	</a>
+</svg>
+
+<main>
+	<p id="main">This is the main content.</p>
+</main>
+```
+
+#### Failed Example 4
+
+The [accessible name][] (from the link's text) does not describe the purpose of the link. The other information available on the page is not [programmatically determined link context][] because it is on a different paragraph.
```

Comment:
```suggestion
The [accessible name][] (from the link's text) does not describe the purpose of the link. The other information available on the page is not [programmatically determined link context][] because it is in a different `p` element.
```